clang/test/SemaSYCL/intel-max-global-work-dim-device.cpp

```diff
@@ -137,19 +311,205 @@ int main() {
+    // CHECK-NEXT:  IntegerLiteral{{.*}}0{{$}}
+
+    h.single_task<class test_kernel13>(TRIFuncObjBad2());
+    // CHECK-LABEL: FunctionDecl {{.*}}test_kernel13
+    // CHECK:       ReqdWorkGroupSizeAttr {{.*}}
+    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
+    // CHECK-NEXT:  value: Int 4
+    // CHECK-NEXT:  IntegerLiteral{{.*}}4{{$}}
+    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
+    // CHECK-NEXT:  value: Int 4
+    // CHECK-NEXT:  IntegerLiteral{{.*}}4{{$}}
+    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
+    // CHECK-NEXT:  value: Int 4
+    // CHECK-NEXT:  IntegerLiteral{{.*}}4{{$}}
+    // CHECK:       SYCLIntelMaxGlobalWorkDimAttr {{.*}}
+    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
+    // CHECK-NEXT:  value: Int 0
+    // CHECK-NEXT:  IntegerLiteral{{.*}}0{{$}}
+
+    // Ignore duplicate attribute with same argument value.
+    h.single_task<class test_kernell4>(
+    // CHECK-LABEL: FunctionDecl {{.*}}test_kernell4 'void ()'
+    // CHECK:       SYCLIntelMaxGlobalWorkDimAttr {{.*}}
+    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
+    // CHECK-NEXT:  value: Int 3
+    // CHECK-NEXT:  IntegerLiteral{{.*}}3{{$}}
+        []() [[intel::max_global_work_dim(3),
+               intel::max_global_work_dim(3)]]{}); // Ok
+
 #ifdef TRIGGER_ERROR
     [[intel::max_global_work_dim(1)]] int Var = 0; // expected-error{{'max_global_work_dim' attribute only applies to functions}}
 
-    h.single_task<class test_kernel6>(
-        []() [[intel::max_global_work_dim(-8)]]{}); // expected-error{{'max_global_work_dim' attribute requires a non-negative integral compile time constant expression}}
+    h.single_task<class test_kernel15>(
+        []() [[intel::max_global_work_dim(-8)]]{}); // expected-error{{'max_global_work_dim' attribute requires integer constant between 0 and 3 inclusive}}
 
-    h.single_task<class test_kernel7>(
-        []() [[intel::max_global_work_dim(3),
+    h.single_task<class test_kernell6>(
+        []() [[intel::max_global_work_dim(3),      // expected-note {{previous attribute is here}}
                intel::max_global_work_dim(2)]]{}); // expected-warning{{attribute 'max_global_work_dim' is already applied with different arguments}}
 
-    h.single_task<class test_kernel8>(TRIFuncObjBad());
+    h.single_task<class test_kernel17>(TRIFuncObjBad3());
+
+    h.single_task<class test_kernel18>(TRIFuncObjBad4());
+
+    h.single_task<class test_kernel19>(TRIFuncObjBad5());
+
+    h.single_task<class test_kernel20>(TRIFuncObjBad6());
+
+    h.single_task<class test_kernel21>(TRIFuncObjBad7());
+
+    h.single_task<class test_kernel22>(TRIFuncObjBad8());
+
+    h.single_task<class test_kernel23>(TRIFuncObjBad9());
+
+    h.single_task<class test_kernel24>(TRIFuncObjBad10());
+
+    h.single_task<class test_kernel25>(TRIFuncObjBad11());
+
+    h.single_task<class test_kernel26>(TRIFuncObjBad12());
+
+    h.single_task<class test_kernel27>(TRIFuncObjBad13());
+
+    h.single_task<class test_kernel28>(TRIFuncObjBad14());
```

Comment:
```suggestion
    h.single_task<class test_kernel17>(TRIFuncObjBad3());
    h.single_task<class test_kernel18>(TRIFuncObjBad4());
    h.single_task<class test_kernel19>(TRIFuncObjBad5());
    h.single_task<class test_kernel20>(TRIFuncObjBad6());
    h.single_task<class test_kernel21>(TRIFuncObjBad7());
    h.single_task<class test_kernel22>(TRIFuncObjBad8());
    h.single_task<class test_kernel23>(TRIFuncObjBad9());
    h.single_task<class test_kernel24>(TRIFuncObjBad10());
    h.single_task<class test_kernel25>(TRIFuncObjBad11());
    h.single_task<class test_kernel26>(TRIFuncObjBad12());
    h.single_task<class test_kernel27>(TRIFuncObjBad13());
    h.single_task<class test_kernel28>(TRIFuncObjBad14());
```
Let's save those newlines for later; I don't think they add a whole lot of value here.